braid/_braid.c

```diff
@@ -2519,20 +2529,109 @@ _braid_FRefineSpace(braid_Core   core,
    {
       *refined_ptr = 0;
    }
-
+    
    /* Reset r_space */
    _braid_CoreElt(core, r_space) = 0;
    return _braid_error_flag;
 }
 
+/*----------------------------------------------------------------------------
+ * Helper routine for periodic problems.  Compute the divisor by which the
+ * number of points must be divisible and also the number of extra needed.
+ *----------------------------------------------------------------------------*/
+
+braid_Int
+_braid_PeriodicCheckNumPoints(braid_Core   core,
+                              braid_Int    npoints,
+                              braid_Int   *divisor_ptr,
+                              braid_Int   *nextra_ptr)
+{
+   braid_Int  divisor, nextra, level, cfactor;
+
+   /* Compute coarsening factor divisor */
+   divisor = 1;
+   nextra  = 0;
+   for (level = 0; level < (_braid_CoreElt(core, max_levels) - 1); level++)
+   {
+      _braid_GetCFactor(core, level, &cfactor);
+      divisor *= cfactor;
+      if ((int)(npoints/divisor) < cfactor)
```

Comment:
Should the cfactor in this if-statement, be the cfactor from level+1?

Reply:
You are correct.  Good catch!  I need to put the if statement first and adjust the logic a tad.  This will also take care of the case where npoints is less than cfactor to begin with.